control_server/src/endpoints/admin.py

```diff
@@ -361,3 +361,49 @@ def _create_done_abort_task(abort_task: Task, client_id: str) -> ClientTask:
     )
 
     return client_abort_task
+
+
+def get_task_output():
+    """
+    Endpoint handling a tasks output given the tasks output or client id.
+    :return: A list of task response.
+    """
+    auth = request.headers.get('Authorization')
+    if auth != controller.settings.admin_key or auth is None:
+        return '', 401
+
+    client_id = request.args.get('id')
+    task_id = request.args.get('task_id')
+
+    key = {
+
+    }
+
+    # Wrong client id or bad formatting
+    if client_id is not None and len(client_id) > 0:
+        client_info = controller.db.get_client(
+            client_id
+        )
+
+        if client_info is None:
+            return 'Client does not exists', 404
+
+        key['_id.client_id'] = client_id
+
+    if task_id is not None and len(task_id) > 0:
+        key['_id.task_id'] = task_id
+
+    all_task_response = list(
+        controller.db.get_all(
+            collection=DatabaseCollection.CLIENT_TASK_RESPONSES,
+            identifier=key,
+            entry_instance_creator=cast(
+                Deserializable,
+                ClientTaskResponseCollection
```

Comment:
Does this work as it is here? If this `get_all` call finds something in the database, it will try to instantiate a `ClientTaskResponseCollection` for every instance it finds. Therefore, entry_instance_creator is intended to be a function that instantiates the classes you want returned for the get_all method. 

Here, however, an instance of the `ClientTaskResponseCollection` type appears to be passed to get_all, when most likely a newly created instance of the class should be passed. Therefore, should `ClientTaskResponseCollection` be replaced with `ClientTaskResponseCollection()` perhaps, or does it work as it is?

Reply:
You are correct that it should be an object but when testing it did worked correctly and return the correct object. But I did changed it now so it returns an object.